mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -18,41 +18,50 @@
  ****************************************************************/
 package org.apache.james.mailbox.backup;
 
+import static org.assertj.core.api.Assertions.assertThat;
 
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.util.Arrays;
-import java.util.HashSet;
+import java.util.List;
 
 import org.apache.james.mailbox.MailboxManager;
 import org.apache.james.mailbox.MailboxSession;
 import org.apache.james.mailbox.MessageManager;
 import org.apache.james.mailbox.backup.ZipAssert.EntryChecks;
+import org.apache.james.mailbox.backup.zip.ZipArchivesLoader;
 import org.apache.james.mailbox.backup.zip.Zipper;
-import org.apache.james.mailbox.extension.PreDeletionHook;
-import org.apache.james.mailbox.inmemory.MemoryMailboxManagerProvider;
+import org.apache.james.mailbox.inmemory.manager.InMemoryIntegrationResources;
+import org.apache.james.mailbox.model.Mailbox;
+import org.apache.james.mailbox.model.MailboxAnnotation;
 import org.apache.james.mailbox.model.MailboxId;
 import org.apache.james.mailbox.model.MailboxPath;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import com.github.fge.lambdas.Throwing;
+import reactor.core.publisher.Mono;
 
 class DefaultMailboxBackupTest implements MailboxMessageFixture {
-
-    private static final HashSet<PreDeletionHook> PRE_DELETION_HOOKS = new HashSet<>();
-
     private static final int BUFFER_SIZE = 4096;
 
     private final ArchiveService archiveService = new Zipper();
+    private final MailArchivesLoader archiveLoader = new ZipArchivesLoader();
 
     private MailboxManager mailboxManager;
     private DefaultMailboxBackup backup;
 
+    private MailboxSession sessionUser;
+    private MailboxSession sessionOtherUser;
+
     @BeforeEach
-    void beforeEach() {
-        mailboxManager = MemoryMailboxManagerProvider.provideMailboxManager(PRE_DELETION_HOOKS);
-        backup = new DefaultMailboxBackup(mailboxManager, archiveService);
+    void beforeEach() throws Exception {
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+        backup = new DefaultMailboxBackup(mailboxManager, archiveService, archiveLoader);
+        sessionUser = mailboxManager.createSystemSession(USER);
+        sessionOtherUser = mailboxManager.createSystemSession(OTHER_USER);
     }
 
     private void createMailBoxWithMessage(MailboxSession session, MailboxPath mailboxPath, MessageManager.AppendCommand... messages) throws Exception {
```

Comment:
`createMailBoxWithMessages`

Reply:
BTW I think we can open a mailbox session from the user in the mailboxpath for such provisionning and thus simplify the overall method signature to `void createMailBoxWithMessage(MailboxPath mailboxPath, MessageManager.AppendCommand... messages) throws Exception`

Reply:
The closer we can do is :
mailboxManager.createSystemSession(mailboxPath.getUser());
But it has a runtime cost, so i think i prefer the current solution.

Reply:
A runtime cost *in tests* ? For in memory tests?

Reply:
Yes,  i though it was used elsewhere. As it's used only in this method your proposition makes sense. Thanks